runelite-client/src/main/java/net/runelite/client/Notifier.java

```diff
@@ -146,23 +161,29 @@ public void notify(String message, TrayIcon.MessageType type)
 				.build());
 		}
 
-		if (runeLiteConfig.enableFlashNotification())
+		if (runeLiteConfig.flashNotification() != FlashNotification.DISABLED)
 		{
 			flashStart = Instant.now();
+			mouseLastPressedMillis = client.getMouseLastPressedMillis();
+			flashNotification = runeLiteConfig.flashNotification();
 		}
 
 		log.debug(message);
 	}
 
 	public void processFlash(final Graphics2D graphics)
 	{
-		if (flashStart == null || client.getGameCycle() % 40 >= 20)
+		if (flashStart == null || client.getGameState() != GameState.LOGGED_IN)
 		{
+			flashStart = null;
 			return;
 		}
-		else if (client.getGameState() != GameState.LOGGED_IN)
+
+		if (client.getGameCycle() % 40 >= 20
```

Comment:
This code is what adds the flash effect, so it makes more sense to check `flashNotification` is one of the flash types instead of not all of the non flash types.